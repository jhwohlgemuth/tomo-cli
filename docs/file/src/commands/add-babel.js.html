<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../">
  <title data-ice="title">src/commands/add-babel.js | tomo-cli</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
<meta name="description" content="A friendly command line tool designed to help create sustainable software using web technology"><meta property="twitter:card" content="summary"><meta property="twitter:title" content="tomo-cli"><meta property="twitter:description" content="A friendly command line tool designed to help create sustainable software using web technology"></head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  <a href="test.html" data-ice="testLink">Test</a>
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
<a style="position:relative; top:3px;" href="https://github.com/jhwohlgemuth/tomo-cli"><img width="20px" src="./image/github.png"></a></header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CommandError">CommandError</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-OfflineWarning">OfflineWarning</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Task">Task</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TaskList">TaskList</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-UI">UI</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Warning">Warning</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-populateQueue">populateQueue</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/BasicEditor.js~BasicEditor.html">BasicEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/MakefileEditor.js~MakefileEditor.html">MakefileEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/Scaffolder.js~Scaffolder.html">Scaffolder</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoExist">allDoExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoExistSync">allDoExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoNotExist">allDoNotExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoNotExistSync">allDoNotExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-format">format</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getBinDirectory">getBinDirectory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getCommandDirectory">getCommandDirectory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-parse">parse</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-someDoExist">someDoExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-someDoExistSync">someDoExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-createJsonEditor">createJsonEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-createModuleEditor">createModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getIntendedInput">getIntendedInput</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getVersions">getVersions</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-install">install</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-verifyRustInstallation">verifyRustInstallation</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-BabelConfigModuleEditor">BabelConfigModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-EslintConfigModuleEditor">EslintConfigModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-PackageJsonEditor">PackageJsonEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-PostcssConfigEditor">PostcssConfigEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-WebpackConfigEditor">WebpackConfigEditor</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/commands/add-babel.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import {BabelConfigModuleEditor, install, PackageJsonEditor} from &apos;../utils&apos;;
import {allDoNotExist, someDoExist} from &apos;../utils/common&apos;;

const BABEL_CORE = [
    &apos;@babel/cli&apos;,
    &apos;@babel/core&apos;,
    &apos;@babel/runtime&apos;
];
const BABEL_PRESETS = [
    &apos;@babel/preset-env&apos;
];
const BABEL_PLUGINS = [
    &apos;@babel/plugin-transform-runtime&apos;,
    &apos;@babel/plugin-proposal-class-properties&apos;,
    &apos;@babel/plugin-proposal-export-default-from&apos;,
    &apos;@babel/plugin-proposal-optional-chaining&apos;
];
const BABEL_REACT_PRESET = [
    &apos;@babel/preset-react&apos;
];
const BABEL_DEPENDENCIES = [
    ...BABEL_CORE,
    ...BABEL_PRESETS,
    ...BABEL_PLUGINS
];
/** @ignore */
export const tasks = [
    {
        text: &apos;Create Babel config file&apos;,
        task: async () =&gt; {
            await (new BabelConfigModuleEditor()).create().commit();
        },
        condition: () =&gt; allDoNotExist(&apos;babel.config.js&apos;, &apos;.babelrc&apos;, &apos;.babelrc.js&apos;)
    },
    {
        text: &apos;Install Babel core, CLI, presets, and plugins&apos;,
        task: ({skipInstall}) =&gt; install(BABEL_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({isNotOffline}) =&gt; isNotOffline &amp;&amp; (!(new PackageJsonEditor()).hasAll(...BABEL_DEPENDENCIES) &amp;&amp; someDoExist(&apos;package.json&apos;))
    },
    {
        text: &apos;Install Babel React preset&apos;,
        task: ({skipInstall}) =&gt; install(BABEL_REACT_PRESET, {dev: true, skipInstall}),
        condition: ({useReact}) =&gt; (useReact &amp;&amp; someDoExist(&apos;package.json&apos;)),
        optional: ({useReact}) =&gt; useReact
    },
    {
        text: &apos;Add React support to Babel configuration file&apos;,
        task: async () =&gt; {
            const presets = [...BABEL_PRESETS, ...BABEL_REACT_PRESET];
            await (new BabelConfigModuleEditor()).extend({presets}).commit();
        },
        condition: ({useReact}) =&gt; (useReact &amp;&amp; someDoExist(&apos;babel.config.js&apos;)),
        optional: ({useReact}) =&gt; useReact
    }
];
export default tasks;</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
