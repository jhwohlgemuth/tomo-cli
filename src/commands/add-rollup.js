import {join} from 'path';
import {
    PackageJsonEditor,
    RollupConfigEditor,
    allDoExist,
    allDoNotExist,
    allDoExistSync,
    debug,
    install,
    uninstall
} from '../api';

const DEPLOY_SCRIPTS = {
    predeploy: 'npm-run-all clean build:es build:css copy:assets',
    deploy: 'echo \"Not yet implemented - now.sh or surge.sh are supported out of the box\" && exit 1'
};
const BUILD_DEPENDENCIES = [
    'cpy-cli',
    'del-cli',
    'npm-run-all'
];
const ROLLUP_DEPENDENCIES = [
    'rollup',
    'rollup-plugin-babel',
    'rollup-plugin-commonjs',
    'rollup-plugin-node-resolve',
    'rollup-plugin-replace',
    'rollup-plugin-terser'
];
/**
 * @type {task[]}
 * @see https://rollupjs.org/guide/en/
 */
export const addRollup = [
    {
        text: 'Create Rollup configuration file',
        task: async ({outputDirectory, sourceDirectory, useReact}) => {
            const input =  `'${sourceDirectory}/main.js${useReact ? 'x' : ''}'`;
            const output = {
                file: `'${outputDirectory}/bundle.min.js'`
            };
            const plugins = [, `commonjs()`]; // the commonjs plugin needs to be the second item in the array
            await (new RollupConfigEditor())
                .create()
                .prepend(`import {terser} from 'rollup-plugin-terser';`)
                .prepend(`import replace from 'rollup-plugin-replace';`)
                .prepend(`import resolve from 'rollup-plugin-node-resolve';`)
                .prepend(`import commonjs from 'rollup-plugin-commonjs';`)
                .prepend(`import babel from 'rollup-plugin-babel';`)
                .prepend(`/* eslint-disable max-len */`)
                .extend({input, output})
                .extend(useReact ? {plugins} : {})
                .commit();
        },
        condition: () => allDoNotExist('webpack.config.js')
    },
    {
        text: 'Add Rollup build tasks to package.json',
        task: async ({assetsDirectory, outputDirectory, sourceDirectory}) => {
            const scripts = {
                ...DEPLOY_SCRIPTS,
                clean: `del-cli ${outputDirectory}`,
                copy: 'npm-run-all --parallel copy:assets copy:index',
                'copy:assets': `cpy \"${assetsDirectory}/!(css)/**/*.*\" \"${assetsDirectory}/**/[.]*\" ${outputDirectory} --parents --recursive`,
                'copy:index': `cpy \"${assetsDirectory}/index.html\" ${outputDirectory}`,
                'prebuild:es': `del-cli ${join(outputDirectory, assetsDirectory)}`,
                'build:es': 'rollup -c',
                'postbuild:es': 'npm run copy',
                'watch:assets': `watch \"npm run copy\" ${assetsDirectory}`,
                'watch:es': `watch \"npm run build:es\" ${sourceDirectory}`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Configure dev task',
        task: async ({skipInstall}) => {
            const scripts = {
                dev: 'stmux [ \"npm run watch:es\" : \"npm run lint:ing\" ]'
            };
            try {
                await install(['stmux'], {dev: true, skipInstall});
            } catch (err) {
                await debug(err, 'Failed to install stmux');
            }
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json', '.eslintrc.js'),
        optional: () => allDoExistSync('package.json', '.eslintrc.js')
    },
    {
        text: 'Install Rollup dependencies',
        task: ({skipInstall}) => install([...BUILD_DEPENDENCIES, ...ROLLUP_DEPENDENCIES], {dev: true, skipInstall}),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json')
    }
];
export const removeRollup = [
    {
        text: 'Delete Rollup configuration file',
        task: async () => {
            await (new RollupConfigEditor())
                .delete()
                .commit();
        },
        condition: () => allDoExist('rollup.config.js')
    },
    {
        text: 'Remove Rollup build tasks from package.json',
        task: async () => {
            const scripts = {
                copy: undefined,
                'copy:assets': undefined,
                'copy:index': undefined,
                'watch:assets': undefined,
                dev: undefined,
                'prebuild:es': undefined,
                'build:es': undefined,
                'postbuild:es': undefined,
                'watch:es': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall Rollup dependencies',
        task: () => uninstall([...BUILD_DEPENDENCIES, ...ROLLUP_DEPENDENCIES, 'stmux']),
        condition: () => allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...ROLLUP_DEPENDENCIES)
    }
];
export default addRollup;