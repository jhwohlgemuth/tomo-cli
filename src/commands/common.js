import {join} from 'path';
import {mkdirp} from 'fs-extra';
import {
    PackageJsonEditor,
    Scaffolder,
    allDoNotExist
} from '../api';

/** @ignore */
export const createEditorConfiguration = [
    {
        text: 'Create editor configuration file',
        task: async ({overwrite}) => {
            await (new Scaffolder(join(__dirname, 'common', 'templates')))
                .overwrite(overwrite)
                .target('.')
                .copy('.editorconfig', '.editorconfig')
                .commit();
        },
        condition: () => allDoNotExist('.editorconfig')
    }
];
/** @ignore */
export const createPackageJson = [
    {
        text: 'Create package.json',
        task: async () => {
            await (new PackageJsonEditor())
                .create()
                .commit();
        },
        condition: () => allDoNotExist('package.json')
    }
];
/** @ignore */
export const createSourceDirectory = [
    {
        text: 'Create source directory',
        task: ({sourceDirectory}) => mkdirp(sourceDirectory),
        condition: ({sourceDirectory}) => allDoNotExist(sourceDirectory)
    }
];