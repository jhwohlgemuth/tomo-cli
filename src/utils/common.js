import {join} from 'path';
import {is} from 'ramda';
import {pathExists, pathExistsSync} from 'fs-extra';
import prettier from 'prettier';

const PRETTIER_OPTIONS = {
    bracketSpacing: false,
    parser: 'json-stringify',
    printWidth: 150,
    tabWidth: 4,
    singleQuote: true
};
const newMap = val => new Map(val);
const joinPath = name => join(process.cwd(), name);
const checkPathExists = name => name |> joinPath |> pathExists;
const checkPathExistsSync = name => name |> joinPath |> pathExistsSync;
export const dict = val => val |> Object.entries |> newMap;
export const isEmptyString = data => typeof data === 'string' && data.length === 0;
export const parse = data => data |> JSON.stringify |> JSON.parse;
export const maybeApply = (val, options) => is(Function)(val) ? val(options) : val;
export const getBinDirectory = path => {
    const [packageDirectory] = path.split('Makefile');
    return `${packageDirectory}node_modules/.bin/`;
};
/**
 * Format input code using Prettier
 * @param {*} [code=''] Code to be formatted
 * @example <caption>Prettier options</caption>
 * {
 *     bracketSpacing: false,
 *     parser: 'json-stringify',
 *     printWidth: 150,
 *     tabWidth: 4,
 *     singleQuote: true
 * }
 * @return {string} Code formatted by Prettier
 */
export const format = (code = {}) => prettier
    .format(JSON.stringify(code), PRETTIER_OPTIONS)
    .replace(/"/g, '');
/**
 * Check that at least one file or files exist
 * @param  {...string} args File or folder path(s)
 * @example
 * // some/folder/
 * //   ├─ foo.js
 * //   └── bar.js
 * const hasFoo = someDoExist('some/folder/foo.js');
 * const hasBaz = someDoExist('some/folder/baz.js');
 * const hasSomething = someDoExist('some/folder/bar.js', 'some/folder/baz.js');
 * console.log(hasFoo); // true
 * console.log(hasBaz); // false
 * console.log(hasSomething); // true
 * @return {boolean} Some files/path do exist (true) or all do not exist (false)
 */
export const someDoExist = async (...args) => {
    const checks = await Promise.all(args.map(checkPathExists));
    return checks.some(Boolean);
};
/**
 * Check that at least one file or files exist (synchronous version of {@link someDoExist})
 * @param  {...string} args File or folder path(s)
 * @return {boolean} Some files/path do exist (true) or all do not exist (false)
 */
export const someDoExistSync = (...args) => args.map(checkPathExistsSync).some(Boolean);
/**
 * Check that all files exist
 * @param  {...string} args File of folder paths
 * @return {boolean} All files/paths exist (true) or do not (false)
 */
export const allDoExist = async (...args) => {
    const checks = await Promise.all(args.map(checkPathExists));
    return checks.every(Boolean);
};
/**
 * Check that all files exist (synchronous version of {@link allDoExist})
 * @param  {...string} args File of folder paths
 * @return {boolean} All files/paths exist (true) or do not (false)
 */
export const allDoExistSync = (...args) => args.map(checkPathExistsSync).every(Boolean);
/**
 * Check that all files do not exist
 * @example
 * // some/folder/
 * //   ├─ foo.js
 * //   └── bar.js
 * const noPackageJson = allDoNotExist('some/folder/package.json');
 * console.log(noPackageJson); // true
 * @param  {...string} args File or folder path(s)
 * @return {boolean} All files/paths do not exist (true) or some do (false)
 */
export const allDoNotExist = async (...args) => {
    const checks = await Promise.all(args.map(checkPathExists));
    return checks.every(val => !val);
};
/**
 * Check that all files do not exist (synchronous version of {@link allDoNotExist})
 * @param  {...string} args File or folder path(s)
 * @return {boolean} All files/paths do not exist (true) or some do (false)
 */
export const allDoNotExistSync = (...args) => args.map(checkPathExistsSync).every(val => !val);