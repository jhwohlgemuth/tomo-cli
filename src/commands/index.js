/* eslint-disable no-magic-numbers */
import {choose, withOptions} from '../utils';
import {
    createPackageJson,
    createSourceDirectory
} from './common';
import addA11y from './add-a11y';
import addBabel from './add-babel';
import {addBrowsersync} from './add-browsersync';
import addEsdoc from './add-esdoc';
import addEslint from './add-eslint';
import addJest from './add-jest';
import addMakefile from './add-makefile';
import addMarionette from './add-marionette';
import {addParcel, removeParcel} from './add-parcel';
import {addPostcss, removePostcss} from './add-postcss';
import {addRollup, removeRollup} from './add-rollup';
import {addWebpack, removeWebpack} from './add-webpack';
import createServer from './create-server';

const createProject = [
    ...createPackageJson,
    ...createSourceDirectory,
    ...addBabel,
    ...addEslint
];
const create = {
    project: createProject,
    app: [
        ...createProject,
        ...addPostcss,
        ...addJest,
        choose({
            default: addMarionette,
            useReact: []// do nothing
        }),
        choose({
            default: addWebpack,
            useRollup: addRollup,
            useParcel: addParcel
        }),
        choose({
            default: addBrowsersync,
            useParcel: []// do nothing
        })
    ],
    server: [
        withOptions({sourceDirectory: '.'}),
        ...createPackageJson,
        ...addEslint,
        ...createServer
    ]
};
const add = {
    a11y: addA11y,
    babel: addBabel,
    browsersync: addBrowsersync,
    esdoc: addEsdoc,
    eslint: [
        ...addBabel,
        ...addEslint
    ],
    jest: [
        ...addBabel,
        ...addJest
    ],
    makefile: addMakefile,
    parcel: [
        ...addBabel,
        ...addParcel
    ],
    postcss: addPostcss,
    rollup: [
        ...addBabel,
        ...addRollup
    ],
    webpack: [
        ...addBabel,
        ...addWebpack
    ]
};
const remove = {
    parcel: removeParcel,
    postcss: removePostcss,
    rollup: removeRollup,
    webpack: removeWebpack
};

export default {
    add,
    remove,
    create,
    new: create// alias for create
};
/**
 * @typedef {Object} task
 * @property {string} text Display text for task
 * @property {function} task Task to execute
 * @property {function} condition Predicate to decide when to execute the task (true) or not (false) - can be async or sync
 * @property {function} [optional] Predicate to decide when to show the task (true) or not (false) - MUST be sync
 */