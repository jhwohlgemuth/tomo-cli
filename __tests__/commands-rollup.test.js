import {fileContents, getDirectoryTree, run, useTemporaryDirectory} from './tomo-test';
import {createPackageJson} from '../src/commands/common';
import addEslint from '../src/commands/add-eslint';
import {addRollup, removeRollup} from '../src/commands/add-rollup';

jest.mock('is-online', () => (async () => true));

describeOnlyOnLinux('Rollup', () => {
    let tempDirectory;
    const skipInstall = true;
    const useReact = true;
    const omit = ['extension', 'path', 'size', 'type'];
    const [setTempDir, cleanupTempDir] = useTemporaryDirectory();
    beforeEach(async () => {
        tempDirectory = await setTempDir();
        process.chdir(tempDirectory);
    });
    afterEach(async () => {
        await cleanupTempDir();
    });
    test('Add support', async () => {
        await run(createPackageJson, {});
        await run(addEslint, {skipInstall});
        await run(addRollup, {skipInstall});
        const tree = getDirectoryTree(tempDirectory, {omit});
        const pkg = fileContents('package.json');
        const rollupConfig = fileContents('rollup.config.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(rollupConfig).toMatchSnapshot();
    });
    test('Add support (+React)', async () => {
        await run(createPackageJson, {});
        await run(addEslint, {skipInstall, useReact});
        await run(addRollup, {skipInstall, useReact});
        const tree = getDirectoryTree(tempDirectory, {omit});
        const pkg = fileContents('package.json');
        const rollupConfig = fileContents('rollup.config.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(rollupConfig).toMatchSnapshot();
    });
    test('Remove support', async () => {
        await run(createPackageJson, {});
        await run(addEslint, {skipInstall});
        await run(addRollup, {skipInstall});
        await run(removeRollup, {skipInstall});
        const tree = getDirectoryTree(tempDirectory, {omit});
        const pkg = fileContents('package.json');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
    });
});