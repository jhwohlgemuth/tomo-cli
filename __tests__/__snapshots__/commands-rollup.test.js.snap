// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rollup Add support (+React) 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: package.json
        },
        {
            name: rollup.config.js
        }
    ]
}
"
`;

exports[`Rollup Add support (+React) 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --ext .js,.jsx --fix\\",
        \\"lint:ing\\": \\"watch \\\\\\"npm run lint\\\\\\" ./source\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"predeploy\\": \\"npm-run-all clean build:es build:css copy:assets\\",
        \\"deploy\\": \\"echo \\\\\\"Not yet implemented - now.sh or surge.sh are supported out of the box\\\\\\" && exit 1\\",
        \\"clean\\": \\"del-cli ./output\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets copy:index\\",
        \\"copy:assets\\": \\"cpy \\\\\\"./assets/!(css)/**/*.*\\\\\\" \\\\\\"./assets/**/[.]*\\\\\\" ./output --parents --recursive\\",
        \\"copy:index\\": \\"cpy \\\\\\"./assets/index.html\\\\\\" ./output\\",
        \\"prebuild:es\\": \\"del-cli output/assets\\",
        \\"build:es\\": \\"rollup -c\\",
        \\"postbuild:es\\": \\"npm run copy\\",
        \\"watch:assets\\": \\"watch \\\\\\"npm run copy\\\\\\" ./assets\\",
        \\"watch:es\\": \\"watch \\\\\\"npm run build:es\\\\\\" ./source\\",
        \\"dev\\": \\"stmux [ \\\\\\"npm run watch:es\\\\\\" : \\\\\\"npm run lint:ing\\\\\\" ]\\"
    }
}
"
`;

exports[`Rollup Add support (+React) 3`] = `
"/* eslint-disable max-len */
import babel from 'rollup-plugin-babel';
import commonjs from 'rollup-plugin-commonjs';
import resolve from 'rollup-plugin-node-resolve';
import replace from 'rollup-plugin-replace';
import {terser} from 'rollup-plugin-terser';

export default {
    input: './source/main.jsx',
    output: {
        file: './output/bundle.min.js',
        format: 'iife',
        sourcemap: 'inline'
    },
    plugins: [
        babel({exclude: 'node_modules/**', runtimeHelpers: true}),
        commonjs(),
        resolve({browser: true}),
        replace({'process.env.NODE_ENV': JSON.stringify('production')}),
        terser()
    ]
};"
`;

exports[`Rollup Add support 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: package.json
        },
        {
            name: rollup.config.js
        }
    ]
}
"
`;

exports[`Rollup Add support 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --ext .js,.jsx --fix\\",
        \\"lint:ing\\": \\"watch \\\\\\"npm run lint\\\\\\" ./source\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"predeploy\\": \\"npm-run-all clean build:es build:css copy:assets\\",
        \\"deploy\\": \\"echo \\\\\\"Not yet implemented - now.sh or surge.sh are supported out of the box\\\\\\" && exit 1\\",
        \\"clean\\": \\"del-cli ./output\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets copy:index\\",
        \\"copy:assets\\": \\"cpy \\\\\\"./assets/!(css)/**/*.*\\\\\\" \\\\\\"./assets/**/[.]*\\\\\\" ./output --parents --recursive\\",
        \\"copy:index\\": \\"cpy \\\\\\"./assets/index.html\\\\\\" ./output\\",
        \\"prebuild:es\\": \\"del-cli output/assets\\",
        \\"build:es\\": \\"rollup -c\\",
        \\"postbuild:es\\": \\"npm run copy\\",
        \\"watch:assets\\": \\"watch \\\\\\"npm run copy\\\\\\" ./assets\\",
        \\"watch:es\\": \\"watch \\\\\\"npm run build:es\\\\\\" ./source\\",
        \\"dev\\": \\"stmux [ \\\\\\"npm run watch:es\\\\\\" : \\\\\\"npm run lint:ing\\\\\\" ]\\"
    }
}
"
`;

exports[`Rollup Add support 3`] = `
"/* eslint-disable max-len */
import babel from 'rollup-plugin-babel';
import commonjs from 'rollup-plugin-commonjs';
import resolve from 'rollup-plugin-node-resolve';
import replace from 'rollup-plugin-replace';
import {terser} from 'rollup-plugin-terser';

export default {
    input: './source/main.js',
    output: {
        file: './output/bundle.min.js',
        format: 'iife',
        sourcemap: 'inline'
    },
    plugins: [
        babel({exclude: 'node_modules/**', runtimeHelpers: true}),
        commonjs({namedExports: {'./node_modules/backbone/backbone.js': ['Model', 'history'],'./node_modules/backbone.marionette/lib/backbone.marionette.js': ['Application', 'View', 'MnObject']}}),
        resolve({browser: true}),
        replace({'process.env.NODE_ENV': JSON.stringify('production')}),
        terser()
    ]
};"
`;

exports[`Rollup Remove support 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore
        },
        {
            name: .eslintrc.js
        },
        {
            name: package.json
        }
    ]
}
"
`;

exports[`Rollup Remove support 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --ext .js,.jsx --fix\\",
        \\"lint:ing\\": \\"watch \\\\\\"npm run lint\\\\\\" ./source\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"predeploy\\": \\"npm-run-all clean build:es build:css copy:assets\\",
        \\"deploy\\": \\"echo \\\\\\"Not yet implemented - now.sh or surge.sh are supported out of the box\\\\\\" && exit 1\\",
        \\"clean\\": \\"del-cli ./output\\"
    }
}
"
`;
